src/workerd/api/urlpattern-standard.c++

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "urlpattern-standard.h"
+
+#include "ada.h"
+
+namespace workerd::api::urlpattern {
+std::optional<URLPattern::URLPatternRegexEngine::regex_type> URLPattern::URLPatternRegexEngine::
+    create_instance(std::string_view pattern, bool ignore_case) {
+  jsg::Lock& js = jsg::Lock::from(v8::Isolate::GetCurrent());
+  jsg::Lock::RegExpFlags flags = jsg::Lock::RegExpFlags::kUNICODE_SETS;
+  if (ignore_case) {
+    flags = static_cast<jsg::Lock::RegExpFlags>(
+        flags | static_cast<int>(jsg::Lock::RegExpFlags::kIGNORE_CASE));
+  }
+
+  return js.tryCatch([&]() -> std::optional<regex_type> {
+    return jsg::JsRef(js, js.regexp(kj::StringPtr(pattern.data(), pattern.size()), flags));
```

Comment:
Is this safe if pattern happens to not be null terminated?

Reply:
Pattern is always null terminated, since it is called from Ada, and the underlying data is std::string.